### PR TITLE
Add test if FSDAX is mapped with the MAP_SYNC flag

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Add test if FSDAX is mapped with the `MAP_SYNC` flag

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: .github/workflows/dax.yml [hunk 1]
```diff
@@ -29,11 +41,21 @@ jobs:
...
           ls -al /dev/dax${UMF_TESTS_DEVDAX_NAMESPACE}
           echo UMF_TESTS_DEVDAX_PATH="/dev/dax${UMF_TESTS_DEVDAX_NAMESPACE}"
           echo UMF_TESTS_DEVDAX_SIZE="$(ndctl list --namespace=namespace${UMF_TESTS_DEVDAX_NAMESPACE} | grep size | cut -d':' -f2 | cut -d',' -f1)"
 
+      - name: Check if the FSDAX exists
+        run: |
+          echo UMF_TESTS_FSDAX_NAMESPACE="${UMF_TESTS_FSDAX_NAMESPACE}"
+          echo UMF_TESTS_FSDAX_PATH="${UMF_TESTS_FSDAX_PATH}"
+          ndctl list --namespace=namespace${UMF_TESTS_FSDAX_NAMESPACE}
+          ls -al /dev/$(echo $UMF_TESTS_FSDAX_PATH | cut -d'/' -f3)
```
Comment (lukaszstolarczuk) [inline .github/workflows/dax.yml:55]: there's a lot of greping here, perhaps you should make it the other way around - set `FSDAX_DEVICE=pmem1` and  use it to build all these vars here plus the full path just for testing below (`UMF_TESTS_FSDAX_PATH=/mnt/${device}/file`)

File: .github/workflows/dax.yml [hunk 2]
```diff
@@ -12,6 +22,8 @@ permissions:
 
 env:
   UMF_TESTS_DEVDAX_NAMESPACE : "0.0"
+  UMF_TESTS_FSDAX_NAMESPACE : "1.0"
```
Comment (lukaszstolarczuk) [inline .github/workflows/dax.yml:25]: you could drop this `UMF_TESTS_` prefix here - these vars are just local in this workflow

File: .github/workflows/dax.yml [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
+# It requires:
+# - a DAX device (e.g. /dev/dax0.0) and
+# - a FSDAX device (e.g. /dev/pmem1)
+# configured and mounted in the OS.
+#
+# The DAX device should be specified using the
 # UMF_TESTS_DEVDAX_PATH and UMF_TESTS_DEVDAX_SIZE environment variables.
+#
+# The FSDAX device should be mounted (e.g. /mnt/pmem1) in the OS
+# and specified using the UMF_TESTS_FSDAX_PATH environment variable.
```
Comment (lukaszstolarczuk) [inline .github/workflows/dax.yml:13]: to be precise, this PATH should point to a file on FSDAX

File: test/common/test_helpers.c [hunk 4]
```diff
@@ -76,3 +80,53 @@ tracePoolCreate(umf_memory_pool_handle_t hUpstreamPool,
...
+
+                // Look for the "sf" flag in VmFlags
+                // marking that memory was mapped
+                // with the MAP_SYNC flag.
+                sf_flag = strstr(VmFlags, "sf");
+            }
+        }
+    }
+
+    // fail if the "sf" flag was not found
```
Comment (lukaszstolarczuk) [inline test/common/test_helpers.c:130]: it does not fail, at this point

File: test/common/test_helpers.c [hunk 5]
```diff
@@ -4,9 +4,13 @@
 // This file contains tests for UMF pool API
```
Comment (lukaszstolarczuk) [inline test/common/test_helpers.c:4]: pls bump the license -2024

File: .github/workflows/dax.yml [hunk 6]
```diff
@@ -68,3 +90,7 @@ jobs:
           UMF_TESTS_DEVDAX_PATH="/dev/dax${UMF_TESTS_DEVDAX_NAMESPACE}"
           UMF_TESTS_DEVDAX_SIZE="$(ndctl list --namespace=namespace${UMF_TESTS_DEVDAX_NAMESPACE} | grep size | cut -d':' -f2 | cut -d',' -f1)"
           ctest -C ${{matrix.build_type}} -R devdax -V
+
+      - name: Run only tests of the file memory provider with FSDAX
```
Comment (lukaszstolarczuk) [inline .github/workflows/dax.yml:94]: can't it be simple named `Run FSDAX tests`...? the name is pretty long

Comment (ldorau) [reply .github/workflows/dax.yml]: Done

File: test/provider_file_memory.cpp [hunk 7]
```diff
@@ -121,6 +121,45 @@ static void test_alloc_failure(umf_memory_provider_handle_t provider,
 
 // TESTS
 
+// Test checking if FSDAX was mapped with the MAP_SYNC flag:
```
[the remark this answers is not in the document]
Comment (ldorau) [reply test/provider_file_memory.cpp:127]: Done

File: .github/workflows/dax.yml (hunk 1 above)
Comment (ldorau) [reply .github/workflows/dax.yml]: Done

File: .github/workflows/dax.yml (hunk 2 above)
Comment (ldorau) [reply .github/workflows/dax.yml]: Done

File: .github/workflows/dax.yml (hunk 3 above)
Comment (ldorau) [reply .github/workflows/dax.yml]: Done

File: test/common/test_helpers.c (hunk 4 above)
Comment (ldorau) [reply test/common/test_helpers.c]: Removed

File: test/common/test_helpers.c (hunk 5 above)
Comment (ldorau) [reply test/common/test_helpers.c]: Done

**Comment by ldorau** [comment]
@bratpiorka @lukaszstolarczuk re-review please

File: .github/workflows/dax.yml (hunk 6 above)
Comment (lukaszstolarczuk) [reply .github/workflows/dax.yml]: thx!

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**